### PR TITLE
Feature/Adding product entity and readme update

### PR DESCRIPTION
**Pull request by Ramchy-J**
-Adding ProductEntity
-Updating README.md to include ProductEntity

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
-.idea/
\ No newline at end of file
+.idea/
+out/
+Main.java
```
Comment (eapg) [inline .gitignore]: Please add a new line at the end of this file

File: Entities/ProductEntity.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package Entities;
+
+import java.util.List;
+
+public class ProductEntity {
+    //Attributes
+    private int productId;
```
Comment (eapg) [inline Entities/ProductEntity.java]: Please avoid the use of primitive data types.  Look for Non-Primitive data types and update the entity.

File: Entities/ProductEntity.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    private String name;
+    private double price;
```
Comment (eapg) [inline Entities/ProductEntity.java:9]: Same as Before

**Review by eapg — COMMENTED** [review]
Please move the UML file to the main root as it will contain the whole project

File: Entities/ProductEntity.java (hunk 2 above)
Comment (eapg) [reply Entities/ProductEntity.java]: Also Create a `BaseEntity` with the attributes that will be common between entities with attributes as `entityStatus`, `createdBy`, `updatedBy`, `createdDate`, `updatedDate`. The others entities will extend from this one.

File: .gitignore (hunk 1 above)
Comment (Ramchy-J) [reply .gitignore]: Done

File: Entities/ProductEntity.java (hunk 2 above)
Comment (Ramchy-J) [reply Entities/ProductEntity.java]: Changes from primitive to non-primitive have been done.

But, creating another entity will be against the stament number 5 from the project requeriments, so we can do it in another PR.

**Comment by Ramchy-J** [comment]
-Taking out the UML
-Changing the name of entity
-Changing primitive to non-primitive

File: .gitignore [hunk 4]
```diff
@@ -1 +1,5 @@
-.idea/
\ No newline at end of file
+.idea/
+out/
+
+Main.java
```
Comment (eapg) [inline .gitignore]: Remove `Main.java` from gitignore.

Comment (Ramchy-J) [reply .gitignore]: At this moment is better to keep it, because is empty. Doesn't make sense to push it

File: Entities/Product.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
+package Entities;
```
Comment (reynoldmorel) [inline Entities/Product.java]: It is not a java convention to use `Pascal Case` naming convention for folders. They should be `Lower Case` instead

File: Entities/Product.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.List;
+
+public class Product {
+    //Attributes
```
Comment (reynoldmorel) [inline Entities/Product.java]: Add a space after line comment. For example: `// Attributes`

File: Entities/Product.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    private Integer productId;
+    private String name;
+    private Double price;
+    private String category;
+    private List <String> ingredients;
+
+
+    //Constructors
```
Comment (reynoldmorel) [inline Entities/Product.java]: ditto

File: Entities/Product.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    public Product(Integer productId, String name, Double price, String category, List<String> ingredients) {
+        this.productId = productId;
+        this.name = name;
+        this.price = price;
+        this.category = category;
+        this.ingredients = ingredients;
+    }
+
+
+    //Methods
```
Comment (reynoldmorel) [inline Entities/Product.java]: ditto

File: Entities/Product.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    public int getProductId() {
```
Comment (reynoldmorel) [inline Entities/Product.java]: Use java core wrapper implementation over primitive data types.

File: Entities/Product.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public double getPrice() {
```
Comment (reynoldmorel) [inline Entities/Product.java]: ditto

File: Entities/Product.java [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.List;
+
+public class Product {
+    //Attributes
+    private Integer productId;
+    private String name;
+    private Double price;
+    private String category;
+    private List <String> ingredients;
```
Comment (reynoldmorel) [inline Entities/Product.java]: Ingredients should not be a list of strings, since the ingredients itself might contain properties you'd need.

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: Entities/Product.java (hunk 5 above)
Comment (Ramchy-J) [reply Entities/Product.java]: done

File: Entities/Product.java (hunk 7 above)
Comment (Ramchy-J) [reply Entities/Product.java]: done

File: Entities/Product.java (hunk 8 above)
Comment (Ramchy-J) [reply Entities/Product.java]: done

File: Entities/Product.java (hunk 9 above)
Comment (Ramchy-J) [reply Entities/Product.java]: done

File: Entities/Product.java (hunk 10 above)
Comment (Ramchy-J) [reply Entities/Product.java]: done

File: Entities/Product.java (hunk 11 above)
Comment (Ramchy-J) [reply Entities/Product.java]: done

**Review by reynoldmorel — APPROVED** [review]

**Review by eapg — APPROVED** [review]

File: Entities/ProductEntity.java (hunk 2 above)
Comment (reynoldmorel) [reply Entities/ProductEntity.java]: Could you create another PR with the base entity first?

**Pull request merged**